src/Assignments/SimpleProblemPage.tsx

```diff
@@ -379,12 +418,15 @@ export const SimpleProblemPage: React.FC<SimpleProblemPageProps> = () => {
                                                 New Version
                                             </Button>
                                         }
-                                        <Button variant='danger'
-                                            tabIndex={0}
-                                            onClick={() => confirmEndVersion()}
-                                        >
-                                            End Exam
-                                        </Button>
+                                        {(attemptsRemaining!==topic.topicAssessmentInfo?.maxGradedAttemptsPerVersion) &&
+                                            <Button variant='danger'
+                                                tabIndex={0}
+                                                onClick={() => confirmEndVersion()}
+                                                disabled={attemptsRemaining===topic.topicAssessmentInfo?.maxGradedAttemptsPerVersion}
```

Comment:
Same thing goes for this disabled

Reply:
oops, that disabled was supposed to GO... replaced by the logic that simply removes the button